### ISSUE TITLE
VPN provider support: NAME OF THE PROVIDER

### ISSUE THREAD

**Issue by Gorkrul** [qdm12/gluetun#1881]
HideMe VPN
https://hide.me/en/

all is within a members area, but as I have access I#d like to help and get any necessary files/configs etc.
I tried the openvpn custom, but failed :(

**Comment by daymr** [comment]
I too would like a solution to this. Hide.me is extremely fast. Any config help would be greatly appreciated

**Comment by kolossboss** [comment]
I would also like to see hide.me support.
I'm using OpenVPN for month without problem. But you need to use the IP of the Hide.me server and not the DNS name.
I would also prefer Wireguard instead of OpenVPN

**Comment by fabiengagne** [comment]
I'm coming here to support this suggestion. I would also like to see native support for hide.me and Wireguard.

**Comment by fkmuck1** [comment]
It would be great if you could support hide.me

**Comment by xonex2020** [comment]
Yes that would be great

**Comment by mohitkumarcool** [comment]
its one of the few vpn providers which supports port fordwarding,it would be great if gluetun supports this also

**Comment by lfilho** [comment]
+1 for hide.me, they're also one of the cheapest ones around, so even better!

**Comment by Condorthefox** [comment]
I would also like to have Hide.me added for gluetun please!

**Comment by gleep52** [comment]
Any progress on this yet?

**Comment by DonAppetito** [comment]
I am a long-time hide.me user and I am pretty happy with it. It's cheap, fast, allows PFW and one of the few providers I trust to be non-logging. Adding support for this provider would be awesome.

**Comment by BippyMiester** [comment]
Still waiting on this to happen. :P

**Comment by mohitkumarcool** [comment]
any update ??

**Comment by treymok** [comment]
update?

**Comment by brunoorsolon** [comment]
+1

**Comment by ivolution** [comment]
+1 from my end.
I also have access to the member area - let me know if I can help with that.
@qdm12 Could you give us an update? Is something in the pipeline? Thank you!

**Comment by brunoorsolon** [comment]
Just in case anyone is wondering, it does work well with the custom setup for OpenVPN, been using it for almost 2 months without problems.
But it would be nice to have built-in support for the hide.me provider instead of relying on the custom implementation.

**Comment by ivolution** [comment]
> Just in case anyone is wondering, it does work well with the custom setup for OpenVPN, been using it for almost 2 months without problems. But it would be nice to have built-in support for the hide.me provider instead of relying on the custom implementation.

I have tried yesterday several hours, but always got error messages. I try to set it up in Docker ("Container Manager") within my Synology NAS, with current DSM OS.
Could you @brunoorsolon share how you made Gluetun work with hide.me? I think of which config files you used from hide.me, which changes you made to it (I have used nslookup to exchange the server name for the IP), the configuration of Gluetun, etc.

**Comment by brunoorsolon** [comment]
> > Just in case anyone is wondering, it does work well with the custom setup for OpenVPN, been using it for almost 2 months without problems. But it would be nice to have built-in support for the hide.me provider instead of relying on the custom implementation.
> 
> I have tried yesterday several hours, but always got error messages. I try to set it up in Docker ("Container Manager") within my Synology NAS, with current DSM OS. Could you [@brunoorsolon](https://github.com/brunoorsolon) share how you made Gluetun work with hide.me? I think of which config files you used from hide.me, which changes you made to it (I have used nslookup to exchange the server name for the IP), the configuration of Gluetun, etc.

Hey, sure. I chatgpted my way into it, but after a little bit of trial and error it worked. 
I didn't need to configure anything besides the docker in the compose, the other services that route through VPN are in the same compose file.

**Download hide.me OpenVPN Configuration**

1. Go to: https://hide.me/en/vpnconfig/
2. Log in with your hide.me account
3. Select configuration:
4. Protocol: OpenVPN UDP (faster) or TCP (more reliable)
5. Server location: Amsterdam, Netherlands, Brazil, etc. (download multiple if desired, but you can only use one at a time)
6. Port: 4433 or 3000 (default, either works)
7. Download the .ovpn file(s)
8. Put the files in a place you can access through Docker

Example files:
- amsterdam1.hideservers.net.ovpn
- br.hideservers.net.ovpn

This is the docker config I'm using

```
  gluetun:
    image: qmcgaw/gluetun:latest
    container_name: gluetun
    restart: unless-stopped
    cap_add:
      - NET_ADMIN
    devices:
      - /dev/net/tun:/dev/net/tun
    ports:
      - "8080:8080"      # Web UI
      - "6881:6881"      # Torrent TCP
      - "6881:6881/udp"  # Torrent UDP
    volumes:
      - /opt/docker/gluetun/config:/gluetun
    environment:
      - VPN_SERVICE_PROVIDER=custom
      - VPN_TYPE=openvpn
      - OPENVPN_CUSTOM_CONFIG=/gluetun/<FILE YOU DOWNLOADED>.ovpn
      - OPENVPN_USER=${OPENVPN_USER}
      - OPENVPN_PASSWORD=${OPENVPN_PASSWORD}
      - TZ=Europe/Amsterdam
      - FIREWALL_VPN_INPUT_PORTS=6881
      - FIREWALL_OUTBOUND_SUBNETS=<YOUR_SUBNET>
      - DNS_SERVER=on
      - DNS_UPSTREAM_RESOLVERS=cloudflare
```

**Comment by ivolution** [comment]
[EDIT @ 22.03.2026]
The guide I previously mentioned was updated in the meantime. Now it includes how to run a Wireguard package in Synology, so you can use the easier usable Wireguard connection. Hide.me provides the file in the member section, ready to download.
[/EDIT]

I have now used ChatGPT as well for "debugging" and can now in the end
- run **Gluetun** and **qBittorrent** in
- a **Container Management Project** (a.k.a. _Docker Compose_) on
- my **Synology Diskstation NAS** using the _current DSM 7.3.2-86009_ and mostly this guide https://drfrankenstein.co.uk/qbittorrent-with-gluetun-vpn-in-container-manager-on-a-synology-nas/ with the following adjustments to the "custom.config" (the ovpn file [_OpenVPN Configuration (ver. 2.4.1+)_ -> _Linux_] one downloads from the [hide.me-Member-Area](https://member.hide.me/en/server-status) and the Compose-YAML:

### OpenVPN Configuration
```dockerfile
...
#nslookup for the individual hide.me domain name given in the original OVPN file
remote ${hide.me Server IP instead of the Domain name} 3000
...
```

and add, for example at the end of the file the following lines

```dockerfile
...
#ignore typical IPv6 pushes from server
pull-filter ignore "tun-ipv6"
pull-filter ignore "ifconfig-ipv6"
pull-filter ignore "route-ipv6"
pull-filter ignore "redirect-gateway ipv6"

#ignore Windows DNS pushes
pull-filter ignore "dhcp-renew"
pull-filter ignore "register-dns"
pull-filter ignore "block-outside-dns"

#modern cipher suite (OpenVPN 2.5/2.6 compatible)
data-ciphers AES-256-GCM:AES-128-GCM:AES-256-CBC
data-ciphers-fallback AES-256-CBC

#if anywhere "cipher" is still needed (mostly not the case)
cipher AES-256-CBC
...
```

Save/copy this file to the gluetun folder you have created when following the guide mentioned and linked above.

### Compose-YAML

I only show you the gluetun part. Adjust the rest of the file as explained in the guide mentioned and linked above.

```dockerfile
...
services:
  gluetun:
    image: qmcgaw/gluetun:latest
    container_name: gluetun
    cap_add:
      - NET_ADMIN
    devices:
      - /dev/net/tun:/dev/net/tun
    ports:
      - 8090:8090/tcp # port for qbittorrent
    volumes:
      - /volume1/docker/gluetun:/gluetun
    environment:
      - PUID=${Your personal PUID} #CHANGE_TO_YOUR_UID
      - PGID=${Your personal PGID} #CHANGE_TO_YOUR_GID
      - TZ=${Your personal time zone} #CHANGE_TO_YOUR_TZ
      - VPN_SERVICE_PROVIDER=custom
      - OPENVPN_CUSTOM_CONFIG=/gluetun/${ovpn file name} #CHANGE_TO_YOUR_OVPN_FILE_NAME
      - VPN_TYPE=openvpn
      - OPENVPN_USER=${hide.me user name} #CHANGE_TO_YOUR_HIDE.ME_USER_NAME
      - OPENVPN_PASSWORD=${hide.me password} #CHANGE_TO_YOUR_HIDE.ME_PASSWORD
      - HTTPPROXY=off #change to on if you wish to enable
      - SHADOWSOCKS=off #change to on if you wish to enable
      - FIREWALL_OUTBOUND_SUBNETS=172.20.0.0/16,${your subnet} #CHANGE_SUBNET_ACCORDING_TO_GUIDE
      - UPDATER_PERIOD=24h
    network_mode: synobridge
    labels:
      - com.centurylinklabs.watchtower.enable=false
    security_opt:
      - no-new-privileges:true
    restart: always
...
```

### Security Note

For my understanding and tests this is a safe implementation. I am no expert though. If experienced readers see some red flags in the configuration, please let me and others know (and explain why).
